internal/pkg/config/matcher_test.go

```diff
@@ -0,0 +1,244 @@
+package config
+
+import (
+	"testing"
+
+	"sigs.k8s.io/yaml"
+)
+
+func TestFuncMatcher(t *testing.T) {
```

Comment:
The first 4 cases seem to be testing the loading, while the other cases are testing the matching. I think those should be separate tests.

Reply:
Split out and low-level tests added.